appservice/src/intTest/groovy/nu/yona/server/DeviceManagementTest.groovy

```diff
@@ -99,8 +103,31 @@ class AddDeviceTest extends AbstractAppServiceIntegrationTest
 		// App activity URL is for the new device
 		registerResponse.responseData._links."yona:appActivity".href == newDevice._links.self.href + "/appActivity/"
 
+		def bobMessagesAfterUpdate = appService.getMessages(bob)
+		assertResponseStatusOk(bobMessagesAfterUpdate)
+		def deviceChangeMessages = bobMessagesAfterUpdate.responseData._embedded?."yona:messages".findAll{ it."@type" == "BuddyDeviceChangeMessage" }
+
+		deviceChangeMessages.size() == 1
+		deviceChangeMessages[0]._links.self != null
+		deviceChangeMessages[0]._links."yona:process" == null // Processing happens automatically these days
+		deviceChangeMessages[0]._links."yona:user".href == bob.buddies[0].user.url
+		deviceChangeMessages[0].message == "User added a device named '$newDeviceName'"
+
+		User bobAfterReload = appService.reloadUser(bob)
+		bobAfterReload.buddies[0].user.devices.size == 2
+		if (bobAfterReload.buddies[0].user.devices[0].name == newDeviceName)
```

Comment:
Yes, certainly. We're doing that in multiple places, but I forgot.
It can be done this way:
```
bobAfterReload.buddies[0].user.devices.collect { it.name } as Set == [firstDeviceName, newDeviceName] as Set
```